lib/private/Files/Cache/CacheAccess.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace OC\Files\Cache;
+
+use OC\FilesMetadata\FilesMetadataManager;
+use OC\SystemConfig;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\Cache\ICacheAccess;
+use OCP\Files\IMimeTypeLoader;
+use OCP\IDBConnection;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Lower level access to the file cache
+ */
+class CacheAccess implements ICacheAccess {
+	public function __construct(
+		private IDBConnection $connection,
+		private SystemConfig $systemConfig,
+		private LoggerInterface $logger,
+		private FilesMetadataManager $metadataManager,
+		private IMimeTypeLoader $mimeTypeLoader,
+	) {
+	}
+
+	private function getQuery(): CacheQueryBuilder {
+		return new CacheQueryBuilder(
+			$this->connection,
+			$this->systemConfig,
+			$this->logger,
+			$this->metadataManager,
+		);
+	}
+
+	public function getByFileIdInStorage(int $fileId, int $storageId): ?CacheEntry {
+		$query = $this->getQuery()->selectFileCache();
+		$query->andWhere($query->expr()->eq('filecache.fileid', $query->createNamedParameter($fileId, IQueryBuilder::PARAM_INT)));
+		$query->andWhere($query->expr()->eq('filecache.storage', $query->createNamedParameter($storageId, IQueryBuilder::PARAM_INT)));
+
+		$row = $query->executeQuery()->fetch();
+		return $row ? Cache::cacheEntryFromData($row, $this->mimeTypeLoader) : null;
```

Comment:
Could this be simplified with the following:
```suggestion
    return $this->getByFileIdsInStorage([$fileId], $storageId);
```

Or is there an optimization in having a specific method for only one id?

Reply:
There is an optimization for the sharding-aware version of this implementation, but for this version it can indeed be simplified.